numpy/core/tests/test_datetime.py

```diff
@@ -20,6 +21,18 @@
     _has_pytz = False
 
 
+
+def assert_array_equiv(x, y, err_msg='', verbose=True):
+    """
+    Raises an AssertionError if two array_like objects are not equivalent.
+
+    Equivalent objects are either equal or each NaN/NaT in each position.
+    """
+    assert_array_compare(lambda x, y: (x == y) | ((x != x) & (y != y)),
+                         x, y, err_msg=err_msg, verbose=verbose,
+                         header='Arrays are not equivalent')
+
```

Comment:
I would agree, but assert_array_equal already defines the opposite for NaN. My guess is that changing assert_array_equal on NaN would break existing tests, but I also don't want to introduce a new inconsistency between NaN and NaT -- that's what I'm trying to fix here! This suggests that the safest path forward is to add a new testing function.

On Wed, Jan 13, 2016 at 9:09 AM, Mark Wiebe <notifications@github.com>
wrote:

>> @@ -20,6 +21,18 @@
>>      _has_pytz = False
>>  
>>  
>> +
>> +def assert_array_equiv(x, y, err_msg='', verbose=True):
>> +    """
>> +    Raises an AssertionError if two array_like objects are not equivalent.
>> +
>> +    Equivalent objects are either equal or each NaN/NaT in each position.
>> +    """
>> +    assert_array_compare(lambda x, y: (x == y) | ((x != x) & (y != y)),
>> +                         x, y, err_msg=err_msg, verbose=verbose,
>> +                         header='Arrays are not equivalent')
>> +
> It would probably be better not to introduce a new method like assert_array_equiv, and rather use the assert_array_equal that's already defined in numpy.testing? https://github.com/numpy/numpy/blob/4c2b198c5a9667fcfef3e7d2b9c6cf198bf624f6/numpy/testing/utils.py#L743
> It probably needs to be updated to handle this NaT change.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/numpy/numpy/pull/7001/files#r49618874